test/PositionToken.ts

```diff
@@ -1,15 +1,116 @@
+        });
+
+        it("Should revert if the token id doesn't exist", async function () {
+            await expect(r.positionToken.ownerOf(0)).to.be.reverted;
+        });
+
+        it("Should be mintable from address designated executive", async function () {
+            await r.positionToken.safeMint(r.addr1.address);
+            expect(await r.positionToken.ownerOf(firstTokenId)).to.equal(firstTokenOwnerAddress);
+        });
+
+        it("Should increment the positionTokenId properly", async function () {
+            await r.positionToken.safeMint(r.addr2.address);
+            expect(await r.positionToken.ownerOf(secondTokenId)).to.equal(secondTokenOwnerAddress);
+        });
+
+        it("Should not be mintable from an address other than LeverageEngine", async function () {
+            expect(r.positionToken).to.not.be.undefined;
+        });
+
+        it("Should fail to burn if not executive", async function () {
+            const burnPromise = r.positionToken.connect(secondTokenOwner).burn(firstTokenId);
+            await expect(burnPromise).to.be.revertedWith("onlyExecutive: Caller is not executive");
+        });
+
+        it("Should not allow positionToken owner to burn positionToken directly. Position unwind required via executive", async function () {
+            const burnPromise = r.positionToken.connect(firstTokenOwner).burn(firstTokenId);
+            await expect(burnPromise).to.be.revertedWith("onlyExecutive: Caller is not executive");
+        });
+
+        it("Should allow executive to burn any token", async function () {
+            await r.positionToken.burn(firstTokenId);
+            await r.positionToken.burn(secondTokenId);
+            /* should revert since token has been burned */
+            await expect(r.positionToken.burn(firstTokenId)).to.be.revertedWith("ERC721: operator query for nonexistent token");
+            await expect(r.positionToken.burn(secondTokenId)).to.be.revertedWith("ERC721: operator query for nonexistent token");
+        });
+
+        it("Should continue to increment id properly after tokens have been burned", async function () {
+            await r.positionToken.safeMint(r.addr3.address);
+            expect(await r.positionToken.ownerOf(thirdTokenId)).to.equal(thirdTokenOwnerAddress);
+        });
+
+        it("Should only allow positionToken owner to transfer ownership", async function () {
+            const safeTransferAsThirdOwner = r.positionToken.connect(thirdTokenOwner)["safeTransferFrom(address,address,uint256)"];
+            await safeTransferAsThirdOwner(
+                thirdTokenOwnerAddress,
+                secondTokenOwnerAddress,
+                thirdTokenId,
+            );
+            expect(await r.positionToken.ownerOf(thirdTokenId)).to.equal(secondTokenOwnerAddress);
+            /* thirdTokenOwner no longer owner of thirdTokenId, should no longer be able to transfer: */
+            await expect(
```

Comment:
nitpick - this check is great but it can be in its own` it()`

Reply:
13430a1f3d4dd28a7f2bfffd59d5c732b7b3d15f

Reply:
13430a1f3d4dd28a7f2bfffd59d5c732b7b3d15f